protocols/v2/subprotocols/common-messages/src/setup_connection.rs

```diff
@@ -84,8 +84,23 @@ impl<'decoder> SetupConnection<'decoder> {
 
                 work_selection && version_rolling
             }
-            // TODO
-            _ => todo!(),
+            Protocol::JobDeclarationProtocol => {
+                let available = available_flags.reverse_bits();
+                let required = required_flags.reverse_bits();
+
+                let requires_async_job_mining_passed = (required >> 31) & 1 > 0;
+                let requires_async_job_mining_self = (available >> 31) & 1 > 0;
+
+                let specific_flags_check =
+                    !requires_async_job_mining_self || requires_async_job_mining_passed;
+                let general_flags_check = (available & required) == required;
+
+                specific_flags_check && general_flags_check
+            }
+            Protocol::TemplateDistributionProtocol | Protocol::JobDistributionProtocol => {
```

Comment:
why is `Protocol::JobDistributionProtocol` on this line?

Reply:
Because for both of these cases, we dont receive any flags so no point of checking for them

Reply:
ah ok, for some reason my brain parsed `Protocol::JobDeclarationProtocol` and `Protocol::JobDistributionProtocol`  as if they were the same thing

which is actually a good opportunity to highlight a potential footgun:

I'm about 95% sure that `JobDistributionProtocol` is just some legacy terminology that we forgot to remove after some spec changes

if we check [stratumprotocol.org/specification](https://stratumprotocol.org/specification/) we'll only find 3 subprotocols:
- Mining
- Job Declaration
- Template Distribution

@Fi3 do you confirm that we should clean up these remaining `JobDistributionProtocol` from the code?

if that's the case, we should create an issue so we can clean this up in the future

Reply:
if we are not going to add it the spec I would remove it

Reply:
@Shourya742 I think as well this shouldn't be part of this PR, or at least it should in its own commits explaining what/why